x-pack/legacy/plugins/snapshot_restore/public/app/lib/authorization/components/with_privileges.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useContext } from 'react';
+
+import { AuthorizationContext, MissingPrivileges } from './authorization_provider';
+
+interface Props {
+  privileges: string | string[];
+  children: (
+    childrenProps: {
+      isLoading: boolean;
+      hasPrivileges: boolean;
+      missingPrivileges: string[] | MissingPrivileges;
+    }
+  ) => JSX.Element;
+}
+
+const hasPrivilege = (missingPrivileges: string[] | undefined) => {
+  if (!Array.isArray(missingPrivileges) || missingPrivileges.length === 0) {
+    return true;
+  }
+  return false;
+};
+
+export const WithPrivileges = ({ privileges, children }: Props) => {
```

Comment:
could we consider renaming this to something like `PrivilegesProvider` or something else that makes it clear that the usage is a HOC?

Reply:
I prefer to keep the "Provider" term for our context Provider and not mix both. As, indeed, it is not an HOC, I renamed it `<AuthPrivileges></AuthPrivileges>`. Let me know what you think